### PR TITLE
ARROW-9020: [Python] read_json won't respect explicit_schema in parse_options

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: python/pyarrow/_json.pyx [hunk 1]
```diff
@@ -91,19 +92,29 @@ cdef class ParseOptions:
     newlines_in_values: bool, optional (default False)
         Whether objects may be printed across multiple lines (for example
         pretty printed). If false, input must end with an empty line.
+    unexpected_field_behavior: str, default "infer"
+        How JSON fields outside of explicit_schema (if given) are treated.
+        Possible behaviors:
+         - "ignore": unexpected JSON fields are ignored
```
Comment (jorisvandenbossche) [inline python/pyarrow/_json.pyx:98]:
```suggestion
        Possible behaviors:

         - "ignore": unexpected JSON fields are ignored
```

(I *think* rst needs a blank line here)

File: python/pyarrow/_json.pyx [hunk 2]
```diff
@@ -91,19 +92,29 @@ cdef class ParseOptions:
...
+         - "error": error out on unexpected JSON fields
+         - "infer": unexpected JSON fields are type-inferred and included in
+                    the output
```
Comment (jorisvandenbossche) [inline python/pyarrow/_json.pyx:101]:
```suggestion
         - "infer": unexpected JSON fields are type-inferred and included in
           the output
```

File: python/pyarrow/_json.pyx [hunk 3]
```diff
@@ -132,6 +143,47 @@ cdef class ParseOptions:
...
+    @property
+    def unexpected_field_behavior(self):
+        """
+        How JSON fields outside of explicit_schema (if given) are treated.
+
+        Possible behaviors:
+         - "ignore": unexpected JSON fields are ignored
+         - "error": error out on unexpected JSON fields
+         - "infer": unexpected JSON fields are type-inferred and included in
+                    the output
```
Comment (jorisvandenbossche) [inline python/pyarrow/_json.pyx:155]: Same formatting comments here

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good to me! Just a few rst formatting comments + the docstring comment below:

In the `explicit_schema` docstring explanation, I would also point to `unexpected_field_behaviour` for how other fields are handled (as right now the "ignores other fields" is confusing)

**Review by pitrou — APPROVED** [review]
LGTM

File: python/pyarrow/_json.pyx (hunk 1 above)
Comment (kszucs) [reply python/pyarrow/_json.pyx]: Updated.

File: python/pyarrow/_json.pyx (hunk 3 above)
Comment (kszucs) [reply python/pyarrow/_json.pyx]: Updated.

File: python/pyarrow/_json.pyx (hunk 2 above)
Comment (kszucs) [reply python/pyarrow/_json.pyx]: Updated.

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request closed without merging**